lib/ansible/modules/cloud/azure/azure_rm_diagnosticsettings.py

```diff
@@ -0,0 +1,388 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Fan Qiu, (fanqiu@microsoft.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_diagnosticsettings
+version_added: "2.9"
+short_description: Manage Azure diagnostic setting.
+description:
+    - Create, updata and delete an diagnostic setting.
+options:
+    name:
+        description:
+            - The name of the diagnostic setting.
+    target:
+        description:
+            - The target which is the diagnostic settings used for.
+            - The identifier of the resource.
+    state:
+        default: present
+        description:
+            -Assert the state of the diagnostic setting. Use C(present) to create or update and C(absent) to delete.
+        choices:
+            - present
+            - absent
+    storage_account_id:
+        description:
+            - The resource ID of the storage account to which you would like to send Diagnostic Logs.
+    service_bus_rule_id:
+        description:
+            - The service bus rule Id of the diagnostic setting.
+            - This is here to maintain backwards compatibility.
+    event_hub_authorization_rule_id:
+        description:
+            - The resource Id for the event hub authorization rule.
+    event_hub_name:
```

Comment:
pls support id also

Reply:
The python sdk needs the name of the event hub, and I think it's the simplest way to provide this information. For other fields, it supports both id or the name + resource_group now.